app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -503,8 +513,18 @@ def vm_instance_defaults
         :volumes    => [new_volume],
         :scsi_controller => { :type => scsi_controller_default_type },
         :datacenter => datacenter,
+        :firmware => 'automatic',
         :boot_order => ['network', 'disk']
       )
     end
+
+    def firmware_mapping(firmware)
+      case firmware
+      when /UEFI/
```

Comment:
I chose the vmware model because I believe it is a better place as the mapping is completely vmware specific. Do you tend to differ?